cohortextractor/tpp_backend.py

```diff
@@ -2353,6 +2353,41 @@ def patients_with_healthcare_worker_flag_on_covid_vaccine_record(self, returning
           patient_id
         """
 
+    def patients_outpatient_appointment_date(
+        self,
+        # attended = None,
+        # is_first_attendance = True,
+        # with_these_treatment_function_codes=tfc_codelist,
+        # with_these_procedure_codes=OPCS_codelist,
+        # referral_request_received_date="YYYY-MM-DD",
+        between=None,
+        returning="binary_flag",
+    ):
+        date_condition, date_joins = self.get_date_condition(
+            "OPA", "Appointment_Date", between
+        )
+
+        if returning == "binary_flag":
+            column_definition = "1"
+        elif returning == "date":
+            column_definition = "MIN(Appointment_Date)"
```

Comment:
other similar functions have parameters like "return first or return last result" to make this more flexible